Aquarius/Src/Aquarius/Sound/SoundBuffer.cpp

```diff
@@ -0,0 +1,91 @@
+// Fix for std::numeric_limits on Windows, necessary to use AudioFile.h
+#define NOMINMAX
+
+#include "SoundBuffer.h"
+#include "Aquarius/Core/Log.h"
+#include <AudioFile.h>
+#include <AL/alext.h>
+#include <AL/al.h>
+#include <string>
+
+
+namespace Aquarius {
+
+    namespace Sound {
+
+        SoundBuffer* SoundBuffer::get()
+        {
+            static SoundBuffer* sound_buffer = new SoundBuffer();
+            return sound_buffer;
+        }
+
+        uint32_t SoundBuffer::addEffect(const std::string file_path)
+        {
+            uint32_t buffer = 0;
+            alGenBuffers(1, &buffer);
+
+            uint32_t format;
+
+            AudioFile<float> sound_file;
+
+            bool loaded = sound_file.load(file_path);
+
+            if(!loaded)
+            {
+                AQ_CORE_ERROR("Unable to open audio file %v, due to %v", file_path);
```

Comment:
May want to return from function here as I wonder if the sound file doesn't load correctly will you be able to call `getNumChannels` and other functions on the AudioFile object below here if it is not loaded